doc: nrf5340audio: clean up headset and gateway on architecture page (#15102)
nrfconnect/sdk-nrf: applications/nrf5340_audio/doc/firmware_architecture.rst

   :local:
   :depth: 2

Each nRF5340 Audio application corresponds to one of two LE Audio modes and one of four specific LE Audio roles:

* For the *connected isochronous stream* (CIS) mode:

  * Unicast client
  * Unicast server

* For the *broadcast isochronous stream* (BIS) mode:

  * Broadcast source
  * Broadcast sink

> Unicast client (CIS) and broadcast source (BIS) receive the audio data from external sources (USB or line input/I2S) and forward it to one or more unicast servers or broadcast sinks, thus becoming source devices for them.

Reviewer: ```suggestion
Unicast client (CIS) and broadcast source (BIS) receive the audio data from external sources (USB or line input/I2S) and forward it to one or more unicast servers or broadcast sinks respectively.
```

> Unicast servers (CIS) and broadcast sinks (BIS) are receiver devices that play back the audio they get from source devices, and will act as earbuds, headphones, speakers, hearing aids, or similar.

Reviewer: This is where we may rephrase a bit. A broadcast sink is always a receiver, but a unicast server can be a sink and a source.

Author: I think this is clarified in line 24?

Unicast servers can also send audio back to the unicast client.
See :ref:`nrf53_audio_app_overview_modes` for more information.

The nRF5340 Audio applications use the same code base, but use different :file:`main.c` files and include different modules and libraries depending on the configuration.

.. important::

>     Before the |NCS| v2.6.0, there was a single nRF53 Audio application with different configuration options for gateway (currently, unicast client and broadcast source) and headset (currently, unicast server and broadcast sink).

Reviewer: ```suggestion
    Before the |NCS| v2.6.0, there was a single nRF53 Audio application with different configuration options for gateway ( unicast client and broadcast source) and headset ( unicast server and broadcast sink).
```

>     For this reason, until it is cleaned up, the code and documentation of the nRF5340 Audio applications still make references to gateway and headset.

Reviewer: Remove this sentence. We may keep these definitions for quite a while longer.

   CIS and BIS mode overview

Connected Isochronous Stream (CIS)
  CIS is a bidirectional (connected) communication protocol that allows for sending separate connected audio streams from a source device to one or more receivers.
  The source can send the audio data using both the left and the right ISO channels at the same time, allowing for stereophonic sound reproduction with synchronized playback.

>   This is the mode available for the unicast applications (:ref:`unicast client<nrf53_audio_unicast_client_app>` as source and :ref:`unicast server<nrf53_audio_unicast_server_app>` as receivers).

Reviewer: ```suggestion
  This is the mode available for the unicast applications (:ref:`unicast client<nrf53_audio_unicast_client_app>` as gateway and :ref:`unicast server<nrf53_audio_unicast_server_app>` as headset).
```

  In the current version of the nRF5340 Audio unicast client, the application offers both unidirectional and bidirectional communication.

>   In the bidirectional communication, the receiver device will send audio from the on-board PDM microphone.

Reviewer: ```suggestion
  In the bidirectional communication, the headset device will send audio from the on-board PDM microphone.
```

  See :ref:`nrf53_audio_app_configuration_select_bidirectional` in the application description for more information.

  You can also enable a walkie-talkie demonstration.

>   In this demonstration, the source device will send audio from the on-board PDM microphone instead of using USB or the line-in.

Reviewer: In this config both devices will be source and sink.

  See :ref:`nrf53_audio_app_configuration_enable_walkie_talkie` in the application description for more information.

Broadcast Isochronous Stream (BIS)
  BIS is a unidirectional (broadcast/disconnected) communication protocol that allows for broadcasting one or more audio streams from a source device to an unlimited number of receivers that are not connected to the source.

>   This is the mode available for the broadcast applications (:ref:`broadcast source<nrf53_audio_broadcast_source_app>` for receivers and :ref:`broadcast sink<nrf53_audio_broadcast_sink_app>` for source devices).

Reviewer: ```suggestion
  This is the mode available for the broadcast applications (:ref:`broadcast source<nrf53_audio_broadcast_source_app>` for gateways and :ref:`broadcast sink<nrf53_audio_broadcast_sink_app>` for headset devices).
```
